praw/reddit.py

```diff
@@ -464,29 +465,39 @@ def info(self, fullnames=None, url=None):
                   different set of submissions.
 
         """
-        if bool(fullnames) == bool(url):
+        none_count = [fullnames, url].count(None)
+        if none_count > 1:
             raise TypeError("Either `fullnames` or `url` must be provided.")
-        if fullnames:
-            if not isinstance(fullnames, list):
-                raise TypeError("fullnames must be a list")
-
-            def generator():
-                for position in range(0, len(fullnames), 100):
-                    fullname_chunk = fullnames[position : position + 100]
-                    params = {"id": ",".join(fullname_chunk)}
+        if none_count < 1:
+            raise TypeError(
+                "Mutually exclusive parameters: `fullnames`, `url`"
+            )
+
+        if fullnames is not None:
+            if isinstance(fullnames, string_types):
+                raise TypeError("`fullnames` must be a non-str iterable.")
+
+            def generator(fullnames):
+                iterable = iter(fullnames)
+                while True:
+                    chunk = list(islice(iterable, 100))
```

Comment:
Is the conversion to using islice needed because some iterables don't support slicing via `[]`?

Reply:
Precisely. Such as generator comprehensions:

```
>>> fullnames = ('t3_{}'.format(id) for id in ('bletrr', 'b8574q'))
>>> fullnames[:1]
[…]
TypeError: 'generator' object is not subscriptable
>>> list(reddit.info(fullnames))
```